nginx.conf

```diff
@@ -167,4 +167,56 @@ http {
             return 403;
         }
     }
+
+      map $uri:$status:$request_method $loggable_module {
+            ~(\.tar\.gz)\:200\:GET$ 1;
+            default 0;
+         }
+
+        map $uri $module {
+            ~/modules/(?<captured_module>[^/]*)(.*(\.tar\.gz)) $captured_module;
+        }
+
+        log_format json escape=json
+            '{'
+            '"time_local":"$time_local",'
+            '"module":"$module",'
+            '"upstream_addr":"$upstream_addr",'
+            '"request":"$request",'
+            '"body_bytes_sent":"$body_bytes_sent",'
+            '"request_time":"$request_time",'
+            '"http_user_agent":"$http_user_agent"'
+            '}';
+
+       server {
+            listen 81;
+            error_page 404  /404.html;
+            set $bucket "cfbs.s3.amazonaws.com";
+
+            location / {
+                #write modules.log if file ends with .tar.gz and response code is 2xx and request method is GET
+                access_log  /home/proxy/modules.log json if=$loggable_module;
+                resolver 8.8.8.8;
```

Comment:
Still don't understand the need for a hardcoded resolver here.

Reply:
the answer is here https://github.com/cfengine/build-website/pull/25#discussion_r740916615

Reply:
https://serverfault.com/questions/240476/how-to-force-nginx-to-resolve-dns-of-a-dynamic-hostname-everytime-when-doing-p/593003#593003